crates/nu-cli/src/reedline_config.rs

```diff
@@ -833,64 +833,76 @@ fn add_parsed_keybinding(
         }
     };
 
-    let keycode = match keybinding
+    let keycode_str = keybinding
         .keycode
         .to_expanded_string("", config)
-        .to_ascii_lowercase()
-        .as_str()
-    {
-        "backspace" => KeyCode::Backspace,
-        "enter" => KeyCode::Enter,
-        c if c.starts_with("char_") => {
-            let mut char_iter = c.chars().skip(5);
-            let pos1 = char_iter.next();
-            let pos2 = char_iter.next();
-
-            let char = if let (Some(char), None) = (pos1, pos2) {
-                char
-            } else {
-                return Err(ShellError::UnsupportedConfigValue {
-                    expected: "char_<CHAR: unicode codepoint>".to_string(),
-                    value: c.to_string(),
-                    span: keybinding.keycode.span(),
-                });
-            };
+        .to_ascii_lowercase();
 
-            KeyCode::Char(char)
-        }
-        "space" => KeyCode::Char(' '),
-        "down" => KeyCode::Down,
-        "up" => KeyCode::Up,
-        "left" => KeyCode::Left,
-        "right" => KeyCode::Right,
-        "home" => KeyCode::Home,
-        "end" => KeyCode::End,
-        "pageup" => KeyCode::PageUp,
-        "pagedown" => KeyCode::PageDown,
-        "tab" => KeyCode::Tab,
-        "backtab" => KeyCode::BackTab,
-        "delete" => KeyCode::Delete,
-        "insert" => KeyCode::Insert,
-        c if c.starts_with('f') => {
-            let fn_num: u8 = c[1..]
-                .parse()
-                .ok()
-                .filter(|num| matches!(num, 1..=20))
-                .ok_or(ShellError::UnsupportedConfigValue {
-                    expected: "(f1|f2|...|f20)".to_string(),
-                    value: format!("unknown function key: {c}"),
+    let keycode = if let Some(rest) = keycode_str.strip_prefix("char_") {
+        let error = |exp: &str, value| ShellError::UnsupportedConfigValue {
+            expected: exp.to_string(),
+            value,
+            span: keybinding.keycode.span(),
+        };
+
+        let mut char_iter = rest.chars();
+        let char = match (char_iter.next(), char_iter.next()) {
+            (Some(char), None) => char,
+            (Some('u'), Some(_)) => {
+                // This will never panic as we know there are at least two symbols
+                let Ok(code_point) = u32::from_str_radix(&rest[1..], 16) else {
```

Comment:
I wonder if we should limit the length of these hex characters to between 2 -8 or something like that? Or does the `char::from_u32(code_point)` cover that scenario already?

Reply:
I thought about this but decided not to implement it because I didn't see much value in doing so. This implementation does not restrict the user in terms of the code format or the number of characters when specifying a character code. For example, the character `A` can be represented as `char_u41` or `char_u000041` with the current implementation. In other words, the number of leading, insignificant characters is not limited since they are not constrained by the `u32::from_str_radix` function itself. The main point is that both conversions, `u32::from_str_radix()` -> `char::from_u32()`, should succeed.